### ISSUE TITLE
Streaming GTCRN → TFLite conversion: slice_scatter + grouped ConvTranspose2d lowering failures

### ISSUE THREAD

**Issue by czoli1976** [Xiaobin-Rong/gtcrn#96]
# Streaming GTCRN → TFLite conversion: slice_scatter + grouped ConvTranspose2d lowering failures

Hi, thanks for publishing GTCRN — really impressive work, especially the streaming ONNX.

We're evaluating GTCRN for integration into a browser-based noise suppression SDK
(currently shipping DTLN). Quality numbers look strong on reverberant and
non-stationary noise, but we're hitting a runtime-efficiency gap and would like
to ship via LiteRT (ex-TFLite) + XNNPACK instead of ONNX Runtime, because the
TFLite path is ~40× more CPU-efficient on this kind of tiny streaming model in
our measurements. Question: have you (or anyone) gotten a TFLite export working
for the streaming variant?

## What we've tried

**1. `onnx2tf` on the published `gtcrn_simple.onnx`:**
fails during shape inference of `MatMul_26` — gets a tensor shape of
`[1, 1, 0, 257]` where the time dim collapses to zero. Looks like a consequence
of the T=1 streaming assumption interacting with a Slice/Reshape that assumes
T≥2 during shape specialization.

**2. `ai-edge-torch` / `litert-torch` (Google's PyTorch → LiteRT path)** — two
sequential failures:

a. `slice_scatter` lowering fails on the in-place cache updates in
   `StreamEncoder.forward` / `StreamDecoder.forward`
   (e.g. `conv_cache[:,:, :2, :] = new_value`). We worked around this by
   writing a functional wrapper class that uses `torch.cat` / `torch.stack`
   for the cache updates instead of in-place assignment. That resolves the
   `slice_scatter` error.

b. After that, conversion fails on a grouped `ConvTranspose2d` in the decoder:

   ```
   expects input feature dimension (16) / feature_group_count = kernel input
   feature dimension (16). Got feature_group_count = 2.
   ```

   This is the `ConvBlock(16, 16, (1,5), stride=(1,2), padding=(0,2), groups=2,
   use_deconv=True, is_last=False)` block. Either MLIR's lowering of grouped
   transposed conv is stricter than PyTorch's (input-channels should equal
   kernel-input-channels × groups), or the implementation is ambiguous for
   this configuration.

## What we'd find helpful

1. **Is there a known-good TFLite export** of the streaming model somewhere?
   Even an internal one we could reference.
2. **Would you consider a small refactor** of `StreamEncoder` / `StreamDecoder`
   to return caches functionally instead of mutating via slice assignment?
   Happy to send a PR — it changes ~20 lines and preserves numerics.
3. **The grouped deconv block**: does it need to be grouped, or would an
   equivalent full (non-grouped) transposed conv work? A split-and-concat
   rewrite should also be equivalent and TFLite-friendly.

Happy to share reproducible scripts (Dockerfile, conversion + measurement
code) if it would help reproduce on your side.

Thanks!

**Comment by Xiaobin-Rong** [comment]
Thank you for your support. You are welcome to submit a PR regarding the TFLite conversion. The following format is expected: a separate folder (e.g., named `stream_tflite`) containing the TFLite file, the running code, and a README document explaining how to use it.

Thanks!
